From e344dc6f9639fa2673d3716342da37c8096dc421 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 30 Apr 2013 15:23:38 +0300
Subject: [PORT FROM MAIN] Telephony: Fix wakeup issues seen due to DataStall recovery

BZ: 28494

Telephony framework uses AlarmManager for recovery mechanism of data
stalled issue. Framework configures the ELAPSED_REALTIME_WAKEUP alarm
for every 6 minutes based on the data state.

This patch modifies the alarm to use ELAPSED_REALTIME which doesn't
wake the device up in sleep state

Change-Id: I7b47ff6e05dfc06f7e49bc859d6e2d746f6b06a3
Orig-Change-Id: I011cff246df3a97e9dd22a2f22e149b771242640
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../telephony/dataconnection/DcTrackerBase.java    |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java b/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
index d5e0ac7..2b707a1 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTrackerBase.java
@@ -1421,7 +1421,7 @@ public abstract class DcTrackerBase extends Handler {
             intent.putExtra(DATA_STALL_ALARM_TAG_EXTRA, mDataStallAlarmTag);
             mDataStallAlarmIntent = PendingIntent.getBroadcast(mPhone.getContext(), 0, intent,
                     PendingIntent.FLAG_UPDATE_CURRENT);
-            mAlarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP,
+            mAlarmManager.set(AlarmManager.ELAPSED_REALTIME,
                     SystemClock.elapsedRealtime() + delayInMs, mDataStallAlarmIntent);
         } else {
             if (VDBG_STALL) {
-- 
1.7.4.1

